pkg/solana/txm/txm.go

```diff
@@ -568,6 +550,82 @@ func (txm *Txm) Enqueue(accountID string, tx *solanaGo.Transaction, txCfgs ...Se
 	return nil
 }
 
+func (txm *Txm) EstimateComputeUnitLimit(ctx context.Context, tx *solanaGo.Transaction) (uint32, error) {
+	res, err := txm.simulateTx(ctx, tx)
+	if err != nil {
+		return 0, err
+	}
+
+	// Return error if response err is non-nil to avoid broadcasting a tx destined to fail
+	if res.Err != nil {
+		sig := solanaGo.Signature{}
+		if len(tx.Signatures) > 0 {
+			sig = tx.Signatures[0]
+		}
+		txm.processSimulationError(uuid.Nil, sig, res)
+		return 0, fmt.Errorf("simulated tx returned error: %v", res.Err)
+	}
+
+	if res.UnitsConsumed == nil || *res.UnitsConsumed == 0 {
+		txm.lggr.Debug("failed to get units consumed for tx")
+		// Do not return error to allow falling back to default compute unit limit
+		return 0, nil
+	}
+
+	unitsConsumed := *res.UnitsConsumed
+
+	// Add buffer to the used compute estimate
+	unitsConsumed = bigmath.AddPercentage(new(big.Int).SetUint64(unitsConsumed), EstimateComputeUnitLimitBuffer).Uint64()
+
+	if unitsConsumed > math.MaxUint32 {
+		txm.lggr.Debug("compute units used with buffer greater than uint32 max", "unitsConsumed", unitsConsumed)
+		// Do not return error to allow falling back to default compute unit limit
+		return 0, nil
+	}
+
+	return uint32(unitsConsumed), nil
+}
+
+func (txm *Txm) simulateTx(ctx context.Context, tx *solanaGo.Transaction) (res *rpc.SimulateTransactionResult, err error) {
```

Comment:
We will `try` to simulate every tx before broadcasting it right? Will this be handled gracefully in terms of performance?

Reply:
simulation before broadcasting should only happen if that is configured - I believe default is `false`